vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,82 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(string_utils::utf_to_wide(base_path));
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home_dir = getenv("HOME");
+    auto env_data = getenv("XDG_DATA_HOME");
+    if (env_data != NULL)
+        root_paths.set_pref_path(string_utils::utf_to_wide(env_data) / fs::path(app_name) / fs::path(app_name));
+
+    root_paths.set_log_path(root_paths.get_pref_path());
+
+    env_data = getenv("XDG_CONFIG_HOME");
+    if (env_data != NULL)
+        root_paths.set_config_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_config_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".config" / fs::path(app_name));
+
+    env_data = getenv("XDG_CACHE_HOME");
+    if (env_data != NULL)
+        root_paths.set_cache_path(string_utils::utf_to_wide(env_data) / fs::path(app_name));
+    else if (env_home_dir != NULL)
+        root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(env_home_dir)) / ".cache" / fs::path(app_name));
+
+    if (env_home_dir != NULL) {
+        // Don't assume that base_path is portable.
+        root_paths.set_shared_path(fs::path(env_home_dir) / ".local/share" / fs::path(app_name));
+    }
+    env_data = getenv("XDG_DATA_DIRS");
+    if (env_data != NULL) {
+        auto env_paths = string_utils::split_string(env_data, ':');
+        for (std::vector<std::string>::iterator iter = env_paths.begin(); iter != env_paths.end(); iter++) {
+            if (fs::exists(*iter / fs::path(app_name))) {
+                root_paths.set_shared_path(string_utils::utf_to_wide(*iter) / fs::path(app_name));
+                break;
+            }
+        }
+    }
+#endif
+
+    // Create default preference and cache path for safety
+    if (!fs::exists(root_paths.get_config_path()))
+        fs::create_directories(root_paths.get_config_path());
+
+    if (!fs::exists(root_paths.get_pref_path()))
+        fs::create_directories(root_paths.get_pref_path());
```

Comment:
This was re-based on last master: 
![Screenshot_2023-10-12_04-21-16](https://github.com/Vita3K/Vita3K/assets/137227028/487c08d7-1120-451e-8a50-a4ac8123fa63)

Unless there's a new commit that hasn't hit yet. What's the commit hash?

Also where's the "appropriate place"? [Macdu](https://github.com/Macdu) asked to put these in a separate function. Is that not wanted? Or am I missing something here?

> It would be nice for all of this to go inside its own function (I think you can put it in app_init.cpp) which would get called here.

Reply:
It's in this PR https://github.com/Vita3K/Vita3K/pull/2989 Sorry.

Reply:
OK the new commit has been rebased, it just took a while for the commit on master to show up on my end.

Reply:
This should have been fixed a few days ago now.